src/jinja2/environment.py

```diff
@@ -898,7 +898,7 @@ def from_string(self, source, globals=None, template_class=None):
     def make_globals(self, d):
         """Return a dict for the globals."""
         if not d:
-            return self.globals
+            return self.globals.copy()
```

Comment:
I know this enforces what the docs say, but is it necessary for the fix? It might save a bit of space if the reference was reused for a project with tons of templates that didn't change the globals.

Reply:
that's true. removed it 👍